scripts/get-courses.py

```diff
@@ -65,67 +89,40 @@ def make_request(endpoint):
             raise RuntimeError(f"Error while parsing JSON object of assignatures/?page={page_num}: {e}")
     else:
         logging.error(f"API call to assignatures/?page={page_num} failed: <{response.status_code}>: {response.text}")
         raise RuntimeError(f"API call to assignatures/?page={page_num} failed: <{response.status_code}>: {response.text}")
-    
+
     page_num += 1
 
 df = pd.DataFrame(courses_json)
 df = df[['id', 'nom', 'credits', 'vigent']]
 df = df[df['vigent'] == 'S']
 df = df.drop(columns=['vigent'])
-df = df.rename(columns={'id': '_id', 'nom': 'name', 'credits': 'creditHours'})
+df = df.rename(columns={'id': 'slug', 'nom': 'name', 'credits': 'creditHours'})
 
 # Filter by courses not present in the database
-courses_in_db = list(mongo_db.courses.find({"_id": {"$in": list(df["_id"])}}, {"_id": 1}))
-courses_in_db = [course["_id"] for course in courses_in_db]
-df = df[~df["_id"].isin(courses_in_db)]
+courses_in_db = list(mongo_db.courses.find({"slug": {"$in": list(df["slug"])}}))
+courses_in_db = [course["slug"] for course in courses_in_db]
+df = df[~df["slug"].isin(courses_in_db)]
 df = df.head(MAX_NEW_COURSES)
 
 logging.info(f"Fetched {len(df)} new courses")
 df.head()
 
-def get_description(id):
-    response = make_request(f'assignatures/{id}/guia')
-
-    if response.ok:
-        try:
-            response_json = json.loads(response.text)
-            if 'descripcio' not in response_json.keys():
-                raise RuntimeError(f"descripcio not in JSON object")
-            return response_json['descripcio']
-        except Exception as e:
-            logging.warning(f"Error while parsing JSON object of assignatures/{id}/guia/: {e}")
-            return None
-    else:
-        logging.warning(f"API call to assignatures/{id}/guia/ failed: <{response.status_code}>: {response.text}")
-        return None
-    
-def make_subject_url(id):
-    if LANGUAGE == "en":
-        return f"https://www.fib.upc.edu/en/studies/bachelors-degrees/bachelor-degree-informatics-engineering/curriculum/syllabus/{id}"
-    elif LANGUAGE == "ca":
-        return f"https://www.fib.upc.edu/ca/estudis/graus/grau-en-enginyeria-informatica/pla-destudis/assignatures/{id}"
-    else:
-        return f"https://www.fib.upc.edu/es/estudios/grados/grado-en-ingenieria-informatica/plan-de-estudios/asignaturas/{id}"
-    
-df['description'] = df['_id'].apply(get_description)
+df['description'] = df['slug'].apply(get_description)
 df = df[df['description'].notnull()]
 
-df['url'] = df['_id'].apply(make_subject_url)
+df['syllabusUrl'] = df['slug'].apply(make_subject_url)
 
 df_json = df.to_dict("records")
-operations = [
-    UpdateOne({'_id': course['_id']}, {'$set': course}, upsert=True)
-    for course in df_json
-]
-result = mongo_db.courses.bulk_write(operations, ordered=False)
-logging.info("Wrote {} new courses, updated {} courses".format(result.upserted_count, result.matched_count))
+
+result = mongo_db.courses.insert_many(df_json)
```

Comment:
There was no point in updating the database entries if you were dropping all the subjects that are already in the database